client/cli/src/params.rs

```diff
@@ -734,6 +734,41 @@ pub struct BuildSpecCmd {
 
 impl_get_log_filter!(BuildSpecCmd);
 
+/// Wrapper type of `String` which holds an arbitary sized unsigned integer formatted as decimal.
+#[derive(Debug, Clone)]
+pub struct BlockNumber(String);
+
+impl FromStr for BlockNumber {
+	type Err = String;
+
+	fn from_str(block_number: &str) -> Result<Self, Self::Err> {
+		if block_number.chars().any(|d| !d.is_digit(10)) {
+			Err(format!(
+				"Invalid block number: {}, expected decimal formatted unsigned integer",
+				block_number
+			))
+		} else {
+			Ok(Self(block_number.to_owned()))
+		}
+	}
+}
+
+impl BlockNumber {
+	/// Wrapper on top of `std::str::parse<N>` but with `Error` as a `String`
+	///
+	/// See `https://doc.rust-lang.org/std/primitive.str.html#method.parse` for more elaborate
+	/// documentation.
+	pub fn parse<N>(&self) -> Result<N, String>
```

Comment:
nothing prevents  parsing `BlockNumber to a signed integer thus it relies on each type's `FromStr` implementation.

For the primitives types `core::num` it will be `Overflow error`